tests/test_merge.py

```diff
@@ -379,6 +381,332 @@ def test_merge(
+        param(
+            (DictConfig(content={}, element_type=str), {"foo": "qux"}),
+            {"foo": "qux"},
+            str,
+            False,
+            id="new_str",
+        ),
+        param(
+            (DictConfig(content={}, element_type=str), {"foo": None}),
+            raises(ValidationError, match="Non optional field"),
+            None,
+            None,
+            id="new_str_none",
+        ),
+        param(
+            (DictConfig(content={}, element_type=str), {"foo": MISSING}),
+            {"foo": MISSING},
+            str,
+            False,
+            id="new_str_missing",
+        ),
+        param(
+            (DictConfig(content={}, element_type=Optional[str]), {"foo": "qux"}),
+            {"foo": "qux"},
+            str,
+            True,
+            id="new_optional_str",
+        ),
+        param(
+            (DictConfig(content={}, element_type=Optional[str]), {"foo": None}),
+            {"foo": None},
+            str,
+            True,
+            id="new_optional_str_none",
+        ),
+        param(
+            (DictConfig(content={}, element_type=Optional[str]), {"foo": MISSING}),
+            {"foo": MISSING},
+            str,
+            True,
+            id="new_optional_str_missing",
+        ),
+        param(
+            (DictConfig(content={"foo": MISSING}, element_type=str), {"foo": "qux"}),
+            {"foo": "qux"},
+            str,
+            False,
+            id="missing_str",
+        ),
+        param(
+            (DictConfig(content={"foo": MISSING}, element_type=str), {"foo": None}),
+            raises(ValidationError, match="Non optional field"),
+            None,
+            None,
+            id="missing_str_none",
+        ),
+        param(
+            (DictConfig(content={"foo": MISSING}, element_type=str), {"foo": MISSING}),
+            {"foo": MISSING},
+            str,
+            False,
+            id="missing_str_missing",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[str]),
+                {"foo": "qux"},
+            ),
+            {"foo": "qux"},
+            str,
+            True,
+            id="missing_optional_str",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[str]),
+                {"foo": None},
+            ),
+            {"foo": None},
+            str,
+            True,
+            id="missing_optional_str_none",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[str]),
+                {"foo": MISSING},
+            ),
+            {"foo": MISSING},
+            str,
+            True,
+            id="missing_optional_str_missing",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": User("Bond")}, element_type=User),
+                {"foo": User("Bond")},
```

Comment:
Can you make it merge a different user name? (ex: `User("007")`)
(same for `optional_user` below)

Reply:
Good idea. Done in https://github.com/omry/omegaconf/pull/749/commits/e81dd715b3b7223e05fe44b38f18aabc18dbe6b9.